### ISSUE TITLE
Problem with CanoFormWithAutGroupOfTable in GAP 4.9

### ISSUE THREAD

**Issue by alex-konovalov** [gap-packages/modisom#2]
The following happens in GAP 4.9.0:
```
gap> A := GroupRing(GF(2), SmallGroup(8,3));
<algebra-with-one over GF(2), with 3 generators>
gap> T := TableByWeightedBasisOfRad(A);
rec( dim := 7, fld := GF(2), rnk := 2, 
  tab := [ [ [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], [ 0*Z(2), 0*Z(2), Z(2)^0, 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0, 0*Z(2), 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0 ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ] ], 
      [ [ 0*Z(2), 0*Z(2), Z(2)^0, Z(2)^0, Z(2)^0, Z(2)^0, Z(2)^0 ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0, Z(2)^0 ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0, 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0 ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ] ],, 
      [ [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0, 0*Z(2), 0*Z(2) ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0, 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), Z(2)^0 ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ], 
          [ 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2), 0*Z(2) ] ] ], wds := [ ,, [ 1, 2 ],, [ 1, 4 ], [ 2, 4 ], [ 1, 6 ] ], 
  wgs := [ 1, 1, 2, 2, 3, 3, 4 ] )
gap> C := CanoFormWithAutGroupOfTable(T);
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `TriangulizedMat' on 1 arguments at /Users/alexk/gap-4.9.0/lib/methsel2.g:250 called from
TriangulizedMat( sub ) at /Users/alexk/gap-4.9.0/lib/meataxe.gi:700 called from
SMTX.InducedActionFactorModuleWithBasis( M, U 
 ) at /Users/alexk/Library/Preferences/GAP/pkg/modisom/gap/cfstab/general.gi:120 called from
BasisSocleSeries( G ) at /Users/alexk/Library/Preferences/GAP/pkg/modisom/gap/cfstab/orbstab.gi:178 called from
HybridMatrixCanoForm( G, U ) at /Users/alexk/Library/Preferences/GAP/pkg/modisom/gap/autiso/autiso.gi:48 called from
ExtendCanoForm( T, i ); at /Users/alexk/Library/Preferences/GAP/pkg/modisom/gap/autiso/autiso.gi:95 called from
...  at *stdin*:10
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> 
```

In GAP 4.8.10 `TableByWeightedBasisOfRad` returns the same `T`, but then there is no error.

**Comment by fingolfin** [comment]
This is caused by a change in GAP 4.9 made to `SMTX.InducedActionFactorModuleWithBasis`. Which BTW is an undocumented function not used by anything in the kernel, which makes it hard to test. But now I see that both autpgrp and modisom use it.

Anyway, in GAP 4.8, it starts like this:
```gap
SMTX.InducedActionFactorModuleWithBasis:=function (module,sub)
local ans, dim, subdim, F,qmodule;

   sub:=List(sub,ShallowCopy);
   TriangulizeMat(sub);
...
```

Which was optimized by me in GAP 4.9 to this:
```gap
SMTX.InducedActionFactorModuleWithBasis:=function(module,sub)
local ans, dim, subdim, F,qmodule;

   sub:=TriangulizedMat(sub);
...
```

Unfortunately, `TriangulizedMat` does not support empty lists. This can be fixed by adding this to the GAP library:
```gap 
InstallOtherMethod( TriangulizedMat, "for an empty list", [ IsList and IsEmpty ],
mat -> []);
```

**Comment by fingolfin** [comment]
This is fixed by https://github.com/gap-system/gap/pull/2260, once that is merged (and also backported to stable-4.9), I see no reason to delay a fresh modisom release.

**Comment by alex-konovalov** [comment]
Fixed - see modisom tests passing at https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.9
